LdapForNet/LdapEntry.cs

```diff
@@ -41,6 +42,77 @@ public LdapEntry ToLdapEntry()
                 Attributes = Attributes.ToDictionary(_ => _.Name, _ => _.GetValues<string>().ToList())
             };
         }
+        
+        public DirectoryAttribute GetAttribute(string attribute) 
+            => this.Attributes.Contains(attribute) ? this.Attributes[attribute] : null;
+
+        private static Guid? GetGuid(byte[] bytes) 
+            => bytes != null && bytes.Length == 16 ? (Guid?) new Guid(bytes) : null;
+
+        public IEnumerable<string> GetObjectClass() => this.GetStrings(LdapAttributes.ObjectClass);
+
+        public IEnumerable<string> GetSubRefs() => this.GetStrings(LdapAttributes.SubRefs);
+
+        public Guid? GetObjectGuid()
+        {
+            var objectGuid = this.GetAttribute(LdapAttributes.ObjectGuid);
+            return objectGuid != null ? GetGuid(objectGuid.GetValue<byte[]>()) : null;
+        }
+
+        public DateTime? GetWhenChanged()
+        {
+            var whenChanged = this.GetString(LdapAttributes.WhenChanged);
+            if (whenChanged != null)
+            {
+                DateTime date = DateTime.ParseExact(whenChanged, "yyyyMMddHHmmss.f'Z'", CultureInfo.InvariantCulture, DateTimeStyles.AssumeLocal);
+                return DateTime.SpecifyKind(date, DateTimeKind.Utc);
+            }
+            return null;
+        }
+        
+        public DateTime? GetModifyTimestamp()
+        {
+            var modifyTimestamp = this.GetString(LdapAttributes.ModifyTimestamp);
```

Comment:
Maybe to create unit tests for DirectoryAttribute ? What do you think?

Reply:
Added some tests, I guess it would be ok for ldap server because it lack of attributes like WhenChanged or ObjectSid.